schema/deploy/shipping/views.sql

```diff
@@ -728,4 +728,53 @@ comment on view shipping.return_results_v2 is
     'Version 2 of view of barcodes and presence/absence results for return of results on website';
 
 
+create or replace view shipping.hcov19_observation_with_presence_absence_result_v1 as
+
+select
+        presence_absence.created::date as sample_tested_date,
+        present as test_result,
+        sample.details->>'sample_origin' as sample_manifest_origin,
+        best_available_encounter_date,
+        age_range_fine,
+        sex,
+        residence_puma,
+        best_available_site,
+        individual,
+        sample.identifier as sample,
+        symptoms,
+        best_available_site_type
+
+    from shipping.sample_with_best_available_encounter_data_v1
+    join warehouse.sample using (sample_id)
+    full outer join shipping.incidence_model_observation_v3 on
+      (sample.identifier = incidence_model_observation_v3.sample)
+    left join warehouse.presence_absence using (sample_id)
+    left join warehouse.target using (target_id)
+    left join warehouse.organism using (organism_id)
+    -- TODO we'll replace this via a replacement join on
+    -- incidence_model_observation_v4 with PUMAs once it's merged to master
+    left join (
+      select
+        encounter_id,
+        tract.hierarchy->'puma' as residence_puma,
+        tract.hierarchy->'neighborhood_district' as residence_neighborhood_district
+      from warehouse.encounter_location
+      left join warehouse.location as address using (location_id)
+      left join warehouse.tract as tract on (address.hierarchy -> 'tract') = tract.identifier
```

Comment:
This join is weird to me. The point of the location.hierarchy is to avoid this kind of double join to location. For any location row at the scale of puma or below, the containing puma should be directly accessible in `hierarchy->'puma'`.

It does look like some records are indeed missing puma in hierarchy and have tract, but I consider this a data loading bug we should (and can) fix.

Note that there's also no guarantee that an encounter location points to an address, so the table alias is slightly misleading.

Reply:
Got it. I'm using `hierarchy->'puma'` now instead of this join.